test/fhirtypes/StructureDefinition.test.ts

```diff
@@ -259,6 +260,255 @@ describe('StructureDefinition', () => {
     });
   });
 
+  describe('#setInstancePropertyByPath', () => {
+    let jsonStructureDefinition: any;
+    let structureDefinition: StructureDefinition;
+    let fooCode: FshCode;
+    let barCode: FshCode;
+    beforeAll(() => {
+      jsonStructureDefinition = defs.findResource('StructureDefinition');
+    });
+    beforeEach(() => {
+      structureDefinition = StructureDefinition.fromJSON(jsonStructureDefinition);
+      observation = StructureDefinition.fromJSON(jsonObservation, structureDefinition);
+      fooCode = new FshCode('foo', 'http://example.com');
+      barCode = new FshCode('bar', 'http://example.com');
+    });
+
+    // Simple values
+    it('should set an instance property which has a value', () => {
+      observation.setInstancePropertyByPath('version', '1.2.3', getResolver(defs));
+      expect(observation.version).toBe('1.2.3');
+    });
+
+    it('should set an instance property which must be created', () => {
+      observation.setInstancePropertyByPath('contextInvariant', 'foo', getResolver(defs));
+      expect(observation.contextInvariant[0]).toBe('foo');
```

Comment:
This is actually testing a few things here -- arrays, arrays with implied index 0, and a property which must be created (which is the purpose of the test).  If there is another property that must be created and is *not* an array, that might be better since it is a more specific "unit".  If not, then I'd suggest adding `[0]` to the path so you're not testing implicit indexes here.